packages/compass-schema/src/components/field/field.jsx

```diff
@@ -134,7 +129,7 @@ class Field extends Component {
   getSemanticType(type) {
     // check if the type represents geo coordinates, if privacy settings allow
 
-    if (this.state.enableMaps) {
+    if (this.props.enableMaps) {
```

Comment:
If you try this out in the UI, it _does_ work, but the change only has an immediate visible effect in one direction because this method actually mutates `type`, so even if `enableMaps` is turned off the type name remains set to `Coordinates`. Doesn’t seem trivial to fix, so I’ll leave my hands off of this :slightly_smiling_face: